### PR TITLE
Remove unused code

### PR DESCRIPTION
**Pull request by amarkpayne**
Suggestions from codacy

### CODE REVIEW

**Comment by amarkpayne** [comment]
This pull request is not finished, but I figured I would do a little bit at a time, and post it here so that I don't do the same thing as someone else (let me know if you are working on this as well). I am mostly following suggestions of codacy.

**Comment by mliu49** [comment]
I hope you don't mind me tacking on related commits removing unused code.

File: rmgpy/molecule/group.py [hunk 1]
```diff
@@ -1366,7 +1366,7 @@ def addExplicitLigands(self):
             #Do not perform is this atom has wildCards
             if atom.hasWildCards: continue
             elif claimedAtomType is atomTypes['CO'] or claimedAtomType is atomTypes['CS']:
-                for atom2, bond12 in atom.bonds.iteritems():
+                for bond12 in atom.bonds.values():
```
Comment (mliu49) [inline rmgpy/molecule/group.py]: You can use `itervalues()` to get a generator object.

File: rmgpy/rmg/model.py [hunk 2]
```diff
@@ -1567,7 +1567,7 @@ def updateUnimolecularReactionNetworks(self):
         # Two partial networks having the same source and containing one or
         # more explored isomers in common must be merged together to avoid
         # double-counting of rates
-        for source, networks in self.networkDict.iteritems():
+        for networks in self.networkDict.values():
```
Comment (mliu49) [inline rmgpy/rmg/model.py]: Also `itervalues()` here.

File: testing/databaseTest.py [hunk 3]
```diff
@@ -557,7 +554,7 @@ def getEndFromBackbone(backbone, endLabels):
 
         # set of all end_labels should be backbone label
         backboneLabel = set([])
-        for end, end_label in endLabels.iteritems():
+        for end_label in endLabels.values():
```
Comment (mliu49) [inline testing/databaseTest.py]: And here.

File: rmgpy/molecule/group.py (hunk 1 above)
Comment (amarkpayne) [inline rmgpy/molecule/group.py]: I'll change these over to generator objects, good catch!

**Comment by amarkpayne** [comment]
@mliu49 I was running into an error running eg1 on this branch, and it appears that it only occurs with the commit `Remove atom label swapping for families that are their own reverse`. With this commit included, running eg1 yields the following error:

```
After model enlargement:
    The model core has 21 species and 39 reactions
    The model edge has 172 species and 532 reactions


Error: Expecting one matching reverse reaction, not zero in reaction family H_Abstraction for forward reaction CC + [CH2]C=C[CH2] <=> C[CH2] + [CH2]C=CC.

Error: There is likely a bug in the RMG-database kinetics reaction family involving a missing group, missing atomlabels, forbidden groups, etc.
Reactant
CC
1 C u0 p0 c0 {2,S} {3,S} {4,S} {5,S}
2 C u0 p0 c0 {1,S} {6,S} {7,S} {8,S}
3 H u0 p0 c0 {1,S}
4 H u0 p0 c0 {1,S}
5 H u0 p0 c0 {1,S}
6 H u0 p0 c0 {2,S}
7 H u0 p0 c0 {2,S}
8 H u0 p0 c0 {2,S}

Reactant
[CH2]C=C[CH2]
multiplicity 3
1  C u0 p0 c0 {2,D} {3,S} {5,S}
2  C u0 p0 c0 {1,D} {4,S} {6,S}
3  C u1 p0 c0 {1,S} {7,S} {8,S}
4  C u1 p0 c0 {2,S} {9,S} {10,S}
5  H u0 p0 c0 {1,S}
6  H u0 p0 c0 {2,S}
7  H u0 p0 c0 {3,S}
8  H u0 p0 c0 {3,S}
9  H u0 p0 c0 {4,S}
10 H u0 p0 c0 {4,S}


[CH2]C=C[CH2]
multiplicity 3
1  C u0 p0 c0 {2,S} {3,D} {5,S}
2  C u1 p0 c0 {1,S} {4,S} {6,S}
3  C u0 p0 c0 {1,D} {7,S} {8,S}
4  C u1 p0 c0 {2,S} {9,S} {10,S}
5  H u0 p0 c0 {1,S}
6  H u0 p0 c0 {2,S}
7  H u0 p0 c0 {3,S}
8  H u0 p0 c0 {3,S}
9  H u0 p0 c0 {4,S}
10 H u0 p0 c0 {4,S}


[CH2]C=C[CH2]
multiplicity 3
1  C u1 p0 c0 {2,S} {3,S} {5,S}
2  C u0 p0 c0 {1,S} {4,D} {6,S}
3  C u1 p0 c0 {1,S} {7,S} {8,S}
4  C u0 p0 c0 {2,D} {9,S} {10,S}
5  H u0 p0 c0 {1,S}
6  H u0 p0 c0 {2,S}
7  H u0 p0 c0 {3,S}
8  H u0 p0 c0 {3,S}
9  H u0 p0 c0 {4,S}
10 H u0 p0 c0 {4,S}

Product
C[CH2]
multiplicity 2
1    C u0 p0 c0 {2,S} {3,S} {4,S} {5,S}
2 *3 C u1 p0 c0 {1,S} {6,S} {7,S}
3    H u0 p0 c0 {1,S}
4    H u0 p0 c0 {1,S}
5    H u0 p0 c0 {1,S}
6    H u0 p0 c0 {2,S}
7    H u0 p0 c0 {2,S}

Product
[CH2]C=CC
multiplicity 2
1     C u0 p0 c0 {2,S} {5,S} {6,S} {7,S}
2     C u0 p0 c0 {1,S} {3,D} {8,S}
3  *1 C u0 p0 c0 {2,D} {4,S} {9,S}
4     C u1 p0 c0 {3,S} {10,S} {11,S}
5     H u0 p0 c0 {1,S}
6     H u0 p0 c0 {1,S}
7     H u0 p0 c0 {1,S}
8     H u0 p0 c0 {2,S}
9  *2 H u0 p0 c0 {3,S}
10    H u0 p0 c0 {4,S}
11    H u0 p0 c0 {4,S}


[CH2]C=CC
multiplicity 2
1     C u0 p0 c0 {2,S} {5,S} {6,S} {7,S}
2     C u1 p0 c0 {1,S} {3,S} {8,S}
3     C u0 p0 c0 {2,S} {4,D} {9,S}
4  *1 C u0 p0 c0 {3,D} {10,S} {11,S}
5     H u0 p0 c0 {1,S}
6     H u0 p0 c0 {1,S}
7     H u0 p0 c0 {1,S}
8     H u0 p0 c0 {2,S}
9     H u0 p0 c0 {3,S}
10    H u0 p0 c0 {4,S}
11 *2 H u0 p0 c0 {4,S}

Error: Debugging why no reaction was found...
Error: Checking whether the family's forbidden species have affected reaction generation...
Traceback (most recent call last):
  File "/home/mark/RMG/RMG-Py/rmgpy/scoop_framework/util.py", line 112, in __call__
    return self.myfn(*args, **kwargs)
  File "/home/mark/RMG/RMG-Py/rmgpy/rmg/react.py", line 85, in reactSpecies
    family.addReverseAttribute(rxn)
  File "/home/mark/RMG/RMG-Py/rmgpy/data/kinetics/family.py", line 1413, in addReverseAttribute
    reactions = self.__generateReactions(rxn.products, products=rxn.reactants, forward=True)
  File "/home/mark/RMG/RMG-Py/rmgpy/data/kinetics/family.py", line 1555, in __generateReactions
    mappingsA = self.__matchReactantToTemplate(moleculeA, template.reactants[0])
  File "/home/mark/RMG/RMG-Py/rmgpy/data/kinetics/family.py", line 1351, in __matchReactantToTemplate
    mappings.extend(reactant.findSubgraphIsomorphisms(child_structure))
AttributeError: 'rmgpy.species.Species' object has no attribute 'findSubgraphIsomorphisms'

Traceback (most recent call last):
  File "rmg.py", line 153, in <module>
    cProfile.runctx(command, global_vars, local_vars, stats_file)
  File "/home/mark/anaconda2/envs/rmg_env/lib/python2.7/cProfile.py", line 49, in runctx
    prof = prof.runctx(statement, globals, locals)
  File "/home/mark/anaconda2/envs/rmg_env/lib/python2.7/cProfile.py", line 140, in runctx
    exec cmd in globals, locals
  File "<string>", line 1, in <module>
  File "/home/mark/RMG/RMG-Py/rmgpy/rmg/main.py", line 734, in execute
    bimolecularReact=self.bimolecularReact)
  File "/home/mark/RMG/RMG-Py/rmgpy/rmg/model.py", line 658, in enlarge
    rxns = reactAll(self.core.species, numOldCoreSpecies, unimolecularReact, bimolecularReact)
  File "/home/mark/RMG/RMG-Py/rmgpy/rmg/react.py", line 377, in reactAll
    rxns = list(react(*spcTuples))
  File "/home/mark/RMG/RMG-Py/rmgpy/rmg/react.py", line 60, in react
    spcTuples)
  File "/home/mark/RMG/RMG-Py/rmgpy/scoop_framework/util.py", line 154, in map_
    return map(WorkerWrapper(args[0]), *args[1:], **kwargs)
  File "/home/mark/anaconda2/envs/rmg_env/lib/python2.7/site-packages/scoop-0.7.2.0-py2.7.egg/scoop/fallbacks.py", line 49, in wrapper
  File "/home/mark/RMG/RMG-Py/rmgpy/scoop_framework/util.py", line 112, in __call__
    return self.myfn(*args, **kwargs)
  File "/home/mark/RMG/RMG-Py/rmgpy/rmg/react.py", line 85, in reactSpecies
    family.addReverseAttribute(rxn)
  File "/home/mark/RMG/RMG-Py/rmgpy/data/kinetics/family.py", line 1413, in addReverseAttribute
    reactions = self.__generateReactions(rxn.products, products=rxn.reactants, forward=True)
  File "/home/mark/RMG/RMG-Py/rmgpy/data/kinetics/family.py", line 1555, in __generateReactions
    mappingsA = self.__matchReactantToTemplate(moleculeA, template.reactants[0])
  File "/home/mark/RMG/RMG-Py/rmgpy/data/kinetics/family.py", line 1351, in __matchReactantToTemplate
    mappings.extend(reactant.findSubgraphIsomorphisms(child_structure))
AttributeError: 'rmgpy.species.Species' object has no attribute 'findSubgraphIsomorphisms'
Coverage.py warning: Module rmg.py was never imported. (module-not-imported)
Makefile:104: recipe for target 'eg1' failed
make: *** [eg1] Error 1
```

Your commit claims that this is legacy code, so perhaps we haven't switched over to a newer methodology here as needed to get rid of this code. How should we proceed? We should probably add a unit test for this as well, as all the checks for this branch passed. Furthermore, it is possible there is an error in the `addReverseAttribute` method as the  `__matchTemplateToReaction` method was sent species objects instead of the expected molecule objects.

In the meantime I am going to remove this commit and place it in a separate remote branch `atomLabelSwapping` until we are ready to add it back in (so that I can test this branch further to make sure we haven't removed too much.)

**Comment by mliu49** [comment]
That's interesting. When I was testing to make sure it was ok to remove, I didn't have any problems with eg1. I'll take another look at it.

**Comment by goldmanm** [comment]
@amarkpayne you know my favorite pull requests to review! These all seem to be excellent changes. Would you be able to rebase, fix conflicts with `liquidTest`, and then I can just double check liquidTest approve and merge it? I should be able to merge after class at 1pm.

**Comment by amarkpayne** [comment]
Certainly! I have resolved the merge conflict and forced pushed to remote.

**Comment by goldmanm** [comment]
@amarkpayne you know my favorite pull requests to review! These all seem to be excellent changes. I just need to double check liquidTest (the only changed file since I previously looked at it) and then I'll merge it.

**Pull request merged**